streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -175,10 +175,10 @@ public Long apply(K aggKey, V value, Long aggregate) {
     /**
      * @return the new sourceName if repartitioned. Otherwise the name of this stream
      */
-    private String repartitionIfRequired(final String storeName) {
+    private String repartitionIfRequired() {
         if (!repartitionRequired) {
             return this.name;
         }
-        return KStreamImpl.createReparitionedSource(this, keySerde, valSerde, storeName);
+        return KStreamImpl.createReparitionedSource(this, keySerde, valSerde);
```

Comment:
`createReparitionedSource` is used for two scenarios: stream aggregation and stream-stream join.

For the latter case since we removed the store name, we can probably just use the operator name before the repartitioning as part of the topic name, which is fine;

But for the first case, the operator name would be `KSTREAM-KEY-SELECT` before the repartition and hence it will not be very information as well. Instead we can treat them differently: for the latter case, still pass in a storeName when calling `KGroupedStreamImpl.repartitionIfRequired` as the `storeSupplier.name()`, which will make the repartition topic name much more informative.